packages/convertim/src/Controller/CustomerController.php

```diff
@@ -36,6 +36,10 @@ public function __construct(
     #[Route('/get-customer-details/{customerUuid}')]
     public function getCustomerDetail(Request $request): Response
     {
+        if (!$this->isConvertimEnabled()) {
+            return $this->convertimNotEnabledResponse();
+        }
+
         if ($this->isProtectedRequest($request) === false) {
             return $this->invalidAuthorizationResponse();
         }
```

Comment:
these two checks are now duplicated in each of the actions, what about creating an event listener that would perform the checks? Something like this: (code generated by chatgpt)

```
namespace App\EventListener;

use Symfony\Component\HttpKernel\Event\ControllerEvent;
use Symfony\Component\HttpKernel\KernelEvents;
use Symfony\Component\EventDispatcher\EventSubscriberInterface;
use App\Controller\AbstractBaseController;

class ControllerCheckListener implements EventSubscriberInterface
{
    public static function getSubscribedEvents()
    {
        return [
            KernelEvents::CONTROLLER => 'onController',
        ];
    }

    public function onController(ControllerEvent $event)
    {
        $controller = $event->getController();

        // Only handle controller as an array (controller class with action method)
        if (is_array($controller) && $controller[0] instanceof AbstractBaseController) {
            $controller[0]->performCheck();
        }
    }
}
```

I am of course open to any other approaches to avoid the DRY violations

Reply:
We discussed it and we are keeping it as it is